src/core/dynamic_system.jl

```diff
@@ -1,60 +1,95 @@
+
 """
-    sesolve(H::AbstractOperator, ψ_0::Ket, t_range::StepRangeLen; e_ops::Vector{AbstractOperator}=(AbstractOperator)[])
-    sesolve(H::Function, ψ_0::Ket, t_range::StepRangeLen; e_ops::Vector{Operator}=(Operator)[])
+    ShrodingerProblem is a structure that is defined to solve the shrodinger equation in time-domain using sesolve(). 
```

Comment:
We have many Operator types and it was very likely (as I learnt the hardware myself) that the user would define the Hamiltonian using one Operator type and define e_ops and c_ops using other (or even mix of other types). This would almost always result in error and crashing of the ODE solver with error messages that was hard to track back to this problem. By defining a "Struct", a user is guaranteed to use proper and uniform operator types and the error message is far more clear. 

BTW, ODE package also has a similar structure.

Reply:
The error messages between passing an incorrect type to a function is exactly the same as passing it to a function, is it not?

If the ODE solver crashed, then the function was too liberal in what it accepted. The same type of assertions in the function definition will have exactly the same effect but without the extra struct.

The ODE function uses a structure since it is used by more than one of their interfaces. Their use case is not the same as ours.

Reply:
As discussed on Teams, we can solve this in a future PR (https://github.com/anyonlabs/Snowflake.jl/issues/242).

Reply:
As discussed in Teams, we will solve this in a future PR (https://github.com/anyonlabs/Snowflake.jl/issues/242)